src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -57,5 +106,152 @@ public void SendAsync_SimpleGet_Success()
                     "Expected cancellation requested on token attached to exception.");
             }
         }
+
+        [Fact]
+        public async void GetAsync_DefaultAutomaticDecompression_ContentDecompressed()
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                using (var client = new HttpClient(handler))
+                {
+                    HttpResponseMessage response = await client.GetAsync(HttpTestServers.RemoteServerGzipUri);
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    Assert.True(responseContent.Contains(DecompressedContentPart));
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async void GetAsync_SetCredential_StatusCodeOK()
+        {
+            Uri uri = HttpTestServers.BasicAuthUriForCreds(Username, Password);
+            using (var handler = new HttpClientHandler())
+            {
+                handler.Credentials = _credential;
+                using (var client = new HttpClient(handler))
+                {
+                    HttpResponseMessage response = await client.GetAsync(uri);
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async void GetAsync_AllowRedirectFalse_StatusCodeRedirect()
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                handler.AllowAutoRedirect = false;
+                using (var client = new HttpClient(handler))
+                {
+                    HttpResponseMessage response = await client.GetAsync(HttpTestServers.RemoteServerRedirectUri);
+                    Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async void GetAsync_AllowRedirectDefault_StatusCodeOK()
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                using (var client = new HttpClient(handler))
+                {
+                    HttpResponseMessage response = await client.GetAsync(HttpTestServers.RemoteServerRedirectUri);
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(6)]
+        public void GetAsync_MaxAutomaticRedirectionsNServerHopsNPlus1_Throw(int hops)
+        {
+            using (var handler = new HttpClientHandler())
+            {
+                handler.MaxAutomaticRedirections = hops;
+                var client = new HttpClient(handler);
+                Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync(HttpTestServers.RedirectUriHops(hops + 1)));
```

Comment:
It is a running test case, I thought, since the exception is being raised, the task is getting executed. OK, will add await on it.